File: src/server/populate_ai2_gallery_new.py
### DOES NOT WORK ###


import json
import requests
import os
import logging


# base_url = "http://localhost:8888"
base_url = "http://gallery-b-dot-mit-appinventor-gallery.appspot.com"

with open('projects.json') as json_data:
    project_list = json.load(json_data)
    for project in project_list:
        project_title = project["aiaPath"].split(".aia")[0]
        aiaPath = os.path.join("uploads", project["aiaPath"])
        imagePath = project["imagePath"].split("api/")[1] if project["imagePath"] else None
        data = {
            "title": project_title,
            "description": project["description"],
            "moreInfo": project["tutorialUrl"],
            "projectId": project["projectId"],
            "email": project["authorUsername"] + "@gmail.com",
            "isFeatured": "true" if "featuredLabel" in project else "false",
        }
        data = {k: v for k, v in data.items() if v is not None}
        print(data)

        # Add project info and create gallery app
        r = requests.post(base_url + "/ode2/receivegalleryprojectinfo", data=data)
        galleryId = r.text
        print(project_title, r.status_code, r.text)

        if r.status_code != 200:
            raise Exception("exception on " + project_title)

        # Add AIA file to gallery app
        with open(aiaPath, "rb") as aiaFile:
            data = aiaFile.read()
        r = requests.post(base_url + "/ode2/receivegalleryproject/" + galleryId, data=data)
        print("added aia for", project_title)

        if r.status_code != 200:
            raise Exception("exception on " + project_title)

        # Add image to gallery app
        if imagePath:
            with open(imagePath, "rb") as imageFile:
                data = imageFile.read()
            r = requests.post(base_url + "/ode2/receivegalleryprojectimage/" + galleryId, data=data)
            print("added image for", project_title)

            if r.status_code != 200:
                raise Exception("exception on " + project_title)
        print('done', project_title)
    

File: src/server/populate_ai2_gallery_new.js
// DOES NOT WORK

const fs = require('fs');
const request = require('request');
const async = require('async');
const path = require('path');

const projects = require('./projects.json');

// const baseUrl = 'http://gallery-b-dot-mit-appinventor-gallery.appspot.com';
const baseUrl = 'http://localhost:8888';

async.eachSeries(projects, (project, cb) => {
  console.log('start ', project.aiaPath);
  const title = project.aiaPath.split('.aia')[0];

  const form = {
    title,
    description: project.description,
    projectId: project.projectId,
    email: `${project.authorUsername}@gmail.com`,
    isFeatured: project.featuredLabel !== null ? 'true' : 'false',
  };

  if (project.tutorialUrl !== null) {
    form.moreInfo = project.tutorialUrl;
  }

  const options = {
    method: 'POST',
    url: `${baseUrl}/ode2/receivegalleryprojectinfo`,
    form,
  };

  console.log(form);
  request(options, (error, response, body) => {
    if (error) throw new Error(error);

    console.log('created gallery id ', body, ' title ', title);
    const galleryId = body;
    fs.createReadStream(`./uploads/${project.aiaPath}`).pipe(request.post(`${baseUrl}/ode2/receivegalleryproject/${galleryId}`));
  });
}, (err) => {
  if (err) throw new Error(err);
  console.log('All done!');
});
